File: design-patterns/java/singleton/StaticBlockSingleton.java
class StaticBlockSingleton {
    // The single instance
    private static StaticBlockSingleton instance;
    
    // Private constructor to prevent instantiation
    private StaticBlockSingleton() {}
    
    // Static block for initialization
    static {
        try {
            instance = new StaticBlockSingleton();
        } catch (Exception e) {
            throw new RuntimeException("Exception occurred in creating singleton instance");
        }
    }
    
    // Public method to get the instance
    public static StaticBlockSingleton getInstance() {
        return instance;
    }
}

File: design-patterns/csharp/singleton/StaticBlockSingleton.cs
using System;

namespace Singleton
{
    public class StaticBlockSingleton
    {
        private static readonly StaticBlockSingleton instance;
        static StaticBlockSingleton()
        {
            try
            {
                instance = new StaticBlockSingleton();
            }
            catch (Exception ex)
            {
                throw new Exception("Exception occurred in creating singleton instance", ex);
            }
        }
        private StaticBlockSingleton() { }
        public static StaticBlockSingleton GetInstance()
        {
            return instance;
        }
    }
} 